stor/dx.py

```diff
@@ -113,96 +165,727 @@ def temp_url(self, lifetime=300, filename=None):
                 URL will be valid
             filename (str, optional): A urlencoded filename to use for
                 attachment, otherwise defaults to object name
+
+        Raises:
+            ValueError: The path points to a project
+            NotFoundError: The path could not be resolved to a file
         """
-        try:
-            if self.canonical_resource:
-                file_handler = dxpy.DXFile(self.canonical_resource)
-                return file_handler.get_download_url(
-                    duration=lifetime,
-                    preauthenticated=True,
-                    filename=filename,
-                    project=self.canonical_project
-                )[0]
-            else:
-                raise DXError('DX Projects cannot have a temporary download url')
-        except ValueError:
-            raise DXError('DXPaths ending in folders cannot have a temporary download url')
+        if not self.resource:
+            raise ValueError('DX Projects cannot have a temporary download url')
+        with _propagate_dx_exceptions():
+            file_handler = dxpy.DXFile(dxid=self.canonical_resource,
+                                       project=self.canonical_project)
+            return file_handler.get_download_url(
+                duration=lifetime,
+                preauthenticated=True,
+                filename=filename,
+                project=self.canonical_project
+            )[0]
 
     @property
     def resource(self):
+        """Returns the resource as a ``PosixPath`` object or None.
```

Comment:
For property, you wouldn't say `Returns`. you should also use an examples section rather than huge block of text. much clearer

```
Returns the virtual or canonical path to the file within the project (as a POSIXPath).

Examples:

   >>> Path('dx://project:dir/file').resource
   PosixPath('/dir/file')
   >>> Path('dx://project-123:file-456').resource
   PosixPath('/file-456')

NOTE: to avoid making API requests, this operation only uses the local string, so `path.name != path.resource.name` for canonical paths.
```

More generally - does it make sense to have it start with a `/`? seems like it's more like a relative path.

In stor and S3, the path after the container/bucket is returned as a relative path, I think that makes more sense here, especially because otherwise this has surprising behavior:

```
local_path = stor.join('/stored', obs_path.resource)
```

which you'd expect to start with `/stored` but instead will start with first path component of OBSPath

Reply:
Changed this, and changed the other scenarios changing this broke.